Ds4 speaker audio base

nickci2002 opened the pull request:
Basic functionality that separates the pad speaker channel from the rest of the audio. This does not include gui components, which need to be implemented.

nickci2002 committed:
Logic update, no QT ui

nickci2002 committed:
Logic update, no QT ui

nickci2002 committed:
Merge branch 'shadps4-emu:main' into DS4-Speaker-Audio-Base

nickci2002 committed:
Fixing errors

rainmakerv3 committed:
Gui boxes

rainmakerv3 committed:
fixes

rainmakerv3 committed:
prevent device list refreshing too fast when game not running

squidbus commented on src/core/libraries/audio/sdl_audio.cpp:
```diff
@@ -21,8 +21,30 @@ class SDLPortBackend : public PortBackend {
+            stream = nullptr;
+            return;
+        } else if (port_name == "Default Device") {
+            devId = SDL_AUDIO_DEVICE_DEFAULT_RECORDING;
```
```suggestion
            devId = SDL_AUDIO_DEVICE_DEFAULT_PLAYBACK;
```

nickci2002 committed:
Fixed Rainmaker's concerns

DanielSvoboda committed:
scePadResetOrientation (#3332)

mikusp committed:
Fix sceAudioOutOutputs (#3335)

marecl committed:
Fixed missing hint (#3334)

georgemoralis committed:
fixed RetrieveNetmask for windows (#3338)

* fixed RetrieveNetmask for windows

* clang..

baggins183 committed:
V_ALIGNBYTE_B32 and V_ALIGNBIT_B32 (#3316)

* implement V_ALIGNBYTE_B32 and V_ALIGNBIT_B32

* fix mask

* uncomment alignbit

kalaposfos13 committed:
Reset orientation on scePadOpen call (#3341)

marecl committed:
Patch is enabled only when game version matches patch version (#3336)

Removed redundancies from CheatsPatches::onSaveButtonClicked()

raphaelthegreat committed:
shader_recompiler: Rework sharp tracking for robustness (#3327)

* shader_recompiler: Remove remnants of old discard

Also constant propagate conditional discard if condition is constant

* resource_tracking_pass: Rework sharp tracking for robustness

* resource_tracking_pass: Add source dominance analysis

When reachability is not enough to prune source list, check if a source dominates all other sources

* resource_tracking_pass: Fix immediate check

How did this work before

* resource_tracking_pass: Remove unused template type

* readlane_elimination_pass: Don't add phi when all args are the same

New sharp tracking exposed some bad sources coming on sampler sharps with aniso disable pattern that also were part of readlane pattern, fix tracking by removing the unnecessary phis inbetween

* resource_tracking_pass: Allow phi in disable aniso pattern

* resource_tracking_pass: Handle not valid buffer sharp and more phi in aniso pattern

mikusp committed:
Sockets are now files (#3319)

* Socket support for read/write/fstat

* Sockets are now files

* Fix ssize_t for windows

* Return posix error codes in net functions

DanielSvoboda committed:
fix patches - Different AppVer (#3344)

* fix patches

* fix linux?

* + mask, mask_jump32 |  - insideMetadata

rainmakerv3 committed:
Multiple controllers: Select active controller and set default controller  (#3169)

* initial commit - not cleanup yet, not usable with imGUI

* Ugly solution to working with ImGUI

* Populate the default controller labels

* Add remove default button

* missing tr calls

* edit imgui flag after updating

* Refactor

* Update sirit

georgemoralis committed:
New Crowdin updates (#3321)

* New translations en_us.ts (Catalan)

* New translations en_us.ts (Norwegian Bokmal)

* New translations en_us.ts (Italian)

* New translations en_us.ts (Swedish)

* New translations en_us.ts (Portuguese, Brazilian)

* New translations en_us.ts (Serbian (Latin))

* New translations en_us.ts (Ukrainian)

* New translations en_us.ts (Russian)

* New translations en_us.ts (Chinese Simplified)

* New translations en_us.ts (Swedish)

* New translations en_us.ts (Turkish)

StevenMiller123 committed:
Net: Fix various socket-related issues (#3347)

* Store platform-specific level in a separate variable

So the level logged in the getsockopt/setsockopt unreachable is actually useful in cases where the level is unknown.

* Define ORBIS_NET_IPPROTO_IPV6

Not implemented yet, but since it's known we might as well add it.

* Fix error codes

Our libSceNet code expects accurate ORBIS_NET_E* errors, while the sys_net code returns ORBIS_NET_ERROR_* errors.

* Remove duplicate getsockname implementation

* Use separate mutex for ReceivePacket calls

Calls to ReceivePacket shouldn't block other socket functions, and allowing them to block these functions frequently causes deadlocks in games that use multiple threads for socket behaviors.
That said, concurrent receives are still a potential issue, so the function should still have a mutex.

* Add missing error codes

* Clang

* Minor nit

Not sure why these were left separate from the rest of the net errnos

* Set __Error() in ConvertReturnErrorCode

Because the new error values are positive, the logic of "negative return is an error" doesn't work anymore. The easiest fix, while retaining corrected error values, is to just set __Error() in ConvertReturnErrorCode, and have that return -1 instead.
I also added some formatting fixes here too.

* Set errno on stubbed P2P socket error returns.

Otherwise the errno is just whatever was set by a previous failing function, which may cause issues in some games.
I used EAGAIN here since it appears to be valid for all three of these functions, but this can be changed if requested.

* Fix missed error returns

* Fix socket methods in file_system

Missed these

mikusp committed:
sys_getpeername (#3354)

georgemoralis committed:
[ci skip] Qt GUI: Update Translation. (#3357)

Co-authored-by: georgemoralis <4313123+georgemoralis@users.noreply.github.com>

rainmakerv3 committed:
controller select crash fix + minor changes (#3355)

* fix crash and ingame controller check

* performance issue fix

* add change controller event to imgui

w1naenator committed:
ime-changes (#3348)

* Changes
-Added support for OrbisImeParamExtended (extended IME parameters) in ImeHandler, ImeState, and ImeUi
-Updated all relevant constructors and logic to propagate and store the extended parameter
- Now fully supports passing extended options from sceImeOpen to the IME UI and backend

* Potential CUSA00434 [Debug] <Critical> assert.cpp:30 assert_fail_debug_msg: Assertion Failed!
buf_len + 1 <= buf_size && "Is your input buffer properly zero-terminated?" at C:/VS/shadPS4-ime-fixes/externals/dear_imgui/imgui_widgets.cpp:4601 fix

* Attempting to resolve an assertion failure in Diablo III:
- Adjusted buffer sizes
- Updated the calculation of text‑length values

---------

Co-authored-by: w1naenator <valdis.bogdans@hotmail.com>

kalaposfos13 committed:
Implement ORBIS_NET_CTL_INFO_HTTP_PROXY_CONFIG (#3366)

rainmakerv3 committed:
Qt: add customizable controller hotkeys (#3369)

* customizable controller hotkeys - initial

* Update input_handler.h

rainmakerv3 committed:
Minor fixes (#3370)

w1naenator committed:
Another ImeDialog hotfix (#3371)

* Changes
-Added support for OrbisImeParamExtended (extended IME parameters) in ImeHandler, ImeState, and ImeUi
-Updated all relevant constructors and logic to propagate and store the extended parameter
- Now fully supports passing extended options from sceImeOpen to the IME UI and backend

* Potential CUSA00434 [Debug] <Critical> assert.cpp:30 assert_fail_debug_msg: Assertion Failed!
buf_len + 1 <= buf_size && "Is your input buffer properly zero-terminated?" at C:/VS/shadPS4-ime-fixes/externals/dear_imgui/imgui_widgets.cpp:4601 fix

* Attempting to resolve an assertion failure in Diablo III:
- Adjusted buffer sizes
- Updated the calculation of text‑length values

* ime-lib another hotfix

Fixed incorrect param->title validation, which caused the IME dialog to fail to appear in Stardew Valley. Need to be checked.

* Clang fix

---------

Co-authored-by: w1naenator <valdis.bogdans@hotmail.com>

DanielSvoboda committed:
Fix Play Time (#3377)

georgemoralis committed:
New Crowdin updates (#3358)

* New translations en_us.ts (Turkish)

* New translations en_us.ts (Spanish)

* New translations en_us.ts (Polish)

* New translations en_us.ts (Albanian)

* New translations en_us.ts (Ukrainian)

* New translations en_us.ts (German)

* New translations en_us.ts (Greek)

* New translations en_us.ts (Vietnamese)

* New translations en_us.ts (Indonesian)

* New translations en_us.ts (Romanian)

* New translations en_us.ts (French)

* New translations en_us.ts (Arabic)

* New translations en_us.ts (Catalan)

* New translations en_us.ts (Danish)

* New translations en_us.ts (Finnish)

* New translations en_us.ts (Hungarian)

* New translations en_us.ts (Italian)

* New translations en_us.ts (Japanese)

* New translations en_us.ts (Korean)

* New translations en_us.ts (Lithuanian)

* New translations en_us.ts (Dutch)

* New translations en_us.ts (Portuguese)

* New translations en_us.ts (Russian)

* New translations en_us.ts (Slovenian)

* New translations en_us.ts (Swedish)

* New translations en_us.ts (Chinese Simplified)

* New translations en_us.ts (Chinese Traditional)

* New translations en_us.ts (Portuguese, Brazilian)

* New translations en_us.ts (Persian)

* New translations en_us.ts (Norwegian Bokmal)

* New translations en_us.ts (Serbian (Latin))

* New translations en_us.ts (Italian)

* New translations en_us.ts (Catalan)

* New translations en_us.ts (Russian)

* New translations en_us.ts (Russian)

* New translations en_us.ts (Arabic)

* New translations en_us.ts (Arabic)

* New translations en_us.ts (Swedish)

* New translations en_us.ts (Portuguese, Brazilian)

* New translations en_us.ts (Portuguese, Brazilian)

* New translations en_us.ts (Norwegian Bokmal)

* New translations en_us.ts (Spanish)

* New translations en_us.ts (Chinese Simplified)

georgemoralis committed:
[ci skip] Qt GUI: Update Translation. (#3378)

Co-authored-by: georgemoralis <4313123+georgemoralis@users.noreply.github.com>

mikusp committed:
Fix get/setsockopt levels (#3384)

mikusp committed:
Stub remaining netctl properties (#3385)

viniciuslrangel committed:
Simple IPC for external control (#3345)

* add simple IPC protocol

to allow communication via stdin/stdout

* ipc: add PATCH_MEMORY command

enables patches & cheates

psucien committed:
libkernel: more network functions for OpenOrbis compatibility (#3373)

StevenMiller123 committed:
libSceAppContent: Use last 16 characters of DLC folder to determine entitlement label (#3375)

* Use last 16 characters of DLC folder name to determine entitlement label

Code comments and commit name say it all.

* Clang

* Adjust comment

ftpdump appends -ac at the end of the entitlement label, so my comment was partially incorrect.

w1naenator committed:
FF9 ImeDialog Hotfix (#3386)

* Changes
-Added support for OrbisImeParamExtended (extended IME parameters) in ImeHandler, ImeState, and ImeUi
-Updated all relevant constructors and logic to propagate and store the extended parameter
- Now fully supports passing extended options from sceImeOpen to the IME UI and backend

* Potential CUSA00434 [Debug] <Critical> assert.cpp:30 assert_fail_debug_msg: Assertion Failed!
buf_len + 1 <= buf_size && "Is your input buffer properly zero-terminated?" at C:/VS/shadPS4-ime-fixes/externals/dear_imgui/imgui_widgets.cpp:4601 fix

* Attempting to resolve an assertion failure in Diablo III:
- Adjusted buffer sizes
- Updated the calculation of text‑length values

* ime-lib another hotfix

Fixed incorrect param->title validation, which caused the IME dialog to fail to appear in Stardew Valley. Need to be checked.

* Clang fix

* FF9 ImeDialog Hotfix

* Removed the validation that disallowed null text and null placeholder, since using null values is valid in `ImeDialog`.
* Added additional debug logs to aid troubleshooting.

---------

Co-authored-by: w1naenator <valdis.bogdans@hotmail.com>

StevenMiller123 committed:
Update file_system.cpp (#3388)

mikusp committed:
Sockets fixes (#3392)

* Print errnos as decimal

* SetSocketOptions: prevent setting SO_TYPE

* Extract net error handler

* Change the local IP retrieval method

* Fix windows

LNDF committed:
video_core: garbage collector (part 1) (#3350)

* Memory information

* Buffer cache GC

* Texture cache GC

* Fix ChangeRegister

* Better image touching

* Buffer async download on GC destroy

* Handle image download, SKIP NON-LINEAR WORKAROUND

* Only download when not dirty

* Correctly handle BDA pagefile update

* Restructure ChangeRegistration

kalaposfos13 committed:
Make libSceRtc fully HLE (#3330)

* Add stubbed libkernel time functions

* Implement remaining stubbed Rtc functions

* Move Rtc from the optionally HLE loading part to the always loaded part

* Remove Rtc from the README list of LLE modules

* Mfw last second hotfix I noticed while double checking that everything's good before opening the PR

marecl committed:
Fixed green artifacts in movies/animations (ffmpeg) (#3398)

* Fixed green artifacts in movies/animations (ffmpeg)

* cleanup&clang

* Avoid for-loop when source and target widths are equal

w1naenator committed:
Ime fixes (#3399)

* Changes
-Added support for OrbisImeParamExtended (extended IME parameters) in ImeHandler, ImeState, and ImeUi
-Updated all relevant constructors and logic to propagate and store the extended parameter
- Now fully supports passing extended options from sceImeOpen to the IME UI and backend

* Potential CUSA00434 [Debug] <Critical> assert.cpp:30 assert_fail_debug_msg: Assertion Failed!
buf_len + 1 <= buf_size && "Is your input buffer properly zero-terminated?" at C:/VS/shadPS4-ime-fixes/externals/dear_imgui/imgui_widgets.cpp:4601 fix

* Attempting to resolve an assertion failure in Diablo III:
- Adjusted buffer sizes
- Updated the calculation of text‑length values

* ime-lib another hotfix

Fixed incorrect param->title validation, which caused the IME dialog to fail to appear in Stardew Valley. Need to be checked.

* Clang fix

* FF9 ImeDialog Hotfix

* Removed the validation that disallowed null text and null placeholder, since using null values is valid in `ImeDialog`.
* Added additional debug logs to aid troubleshooting.

* IME Fixes
- Add missing flags to `OrbisImeExtOption`
- Improve debug logging
- Resolve nonstop `sceImeKeyboardOpen` calls in Stardew Valley (MonoGame engine) for `userId = 254`

---------

Co-authored-by: w1naenator <valdis.bogdans@hotmail.com>

mikusp committed:
select (#3353)

* select

* select for windows

* fixed windows only function

* windows error converts

* fixed up

* fixed compile

* another implementation for windows

* draft rewrite for windows

* implementation for windows

* added some debugging info

* more debugging

* extensive log

* Windows: Add device files to output fd_sets

Cyberpunk 2077 breaks without this.
Also added some formatting changes, to bring the implementation closer to our typical coding standards.

* Clang

* Formatting + cleanup

Removed some excessive logging used for debugging, and did some cleanup on the non-Windows implementation.

---------

Co-authored-by: georgemoralis <giorgosmrls@gmail.com>
Co-authored-by: Stephen Miller <millerste004@gmail.com>

raphaelthegreat committed:
video_core: Rework tile manager (#3374)

* video_core: Rework detiling

* video_core: Support tiling and macrotile detiling

* clang format

* image_info: Cleanups

* resource: Revert some changes

* texture_cache: Fix small error

* image_info: Set depth flag on depth promote

* buffer_cache: Remove level check

* tile_manager: Handle case of staging buffer causing flush

* image_info: Add 2D thick array mode

* image_info: Add slices to mip size

* tile_manager: Set bank swizzle

* buffer_cache: Support image copies from DmaData

* vk_rasterizer: Accelerate trivial render target copies with compute

Before tiling PR compute image copies were done with the following sequence

vkCmdCopyImageToBuffer (in SynchronizeBufferFromImage)  -> vkCmdDispatch (copy) -> vkCmdCopyBufferToImage (in RefreshImage)

With the tiling PR it added extra tiling/detiling steps

vkCmdCopyImageToBuffer -> vkCmdDispatch (tiling) -> vkCmdDispatch (copy) -> vkCmdDispatch (detiling) -> vkCmdCopyBufferToImage

This is quite a bit of overhead for a simple image copy. This commit tries to detect trivial image copies i.e cs shaders that copy the full source image to all of the destination.
So now all this sequence is just a vkCmdCopyImage. How much it triggers depends on the guest

* texture_cache: Fix build

* image: Copy all subresources with buffer too

squidbus committed:
vk_instance: Temporarily disable maintenance8 for MoltenVK

Working on getting fix in for image copy issue.

DanielSvoboda committed:
SystemGesture (#3382)

* SystemGesture

* only STUBBED

georgemoralis committed:
New Crowdin updates (#3380)

* New translations en_us.ts (Turkish)

* New translations en_us.ts (Spanish)

* New translations en_us.ts (Polish)

* New translations en_us.ts (Albanian)

* New translations en_us.ts (Ukrainian)

* New translations en_us.ts (German)

* New translations en_us.ts (Greek)

* New translations en_us.ts (Vietnamese)

* New translations en_us.ts (Indonesian)

* New translations en_us.ts (Romanian)

* New translations en_us.ts (French)

* New translations en_us.ts (Arabic)

* New translations en_us.ts (Catalan)

* New translations en_us.ts (Danish)

* New translations en_us.ts (Finnish)

* New translations en_us.ts (Hungarian)

* New translations en_us.ts (Italian)

* New translations en_us.ts (Japanese)

* New translations en_us.ts (Korean)

* New translations en_us.ts (Lithuanian)

* New translations en_us.ts (Dutch)

* New translations en_us.ts (Portuguese)

* New translations en_us.ts (Russian)

* New translations en_us.ts (Slovenian)

* New translations en_us.ts (Swedish)

* New translations en_us.ts (Chinese Simplified)

* New translations en_us.ts (Chinese Traditional)

* New translations en_us.ts (Portuguese, Brazilian)

* New translations en_us.ts (Persian)

* New translations en_us.ts (Norwegian Bokmal)

* New translations en_us.ts (Serbian (Latin))

* New translations en_us.ts (Italian)

* New translations en_us.ts (Catalan)

* New translations en_us.ts (Russian)

* New translations en_us.ts (Portuguese, Brazilian)

* New translations en_us.ts (Ukrainian)

* New translations en_us.ts (Norwegian Bokmal)

* New translations en_us.ts (Russian)

* New translations en_us.ts (Urdu (Pakistan))

* New translations en_us.ts (Urdu (Pakistan))

* New translations en_us.ts (Urdu (Pakistan))

* New translations en_us.ts (Chinese Simplified)

* New translations en_us.ts (Urdu (Pakistan))

* New translations en_us.ts (Urdu (Pakistan))

* New translations en_us.ts (Urdu (Pakistan))

* New translations en_us.ts (Swedish)

DanielSvoboda committed:
SystemGesture - Rename files to match naming convention (#3403)

StevenMiller123 committed:
libSceAppContent: Determine entitlement labels from additional content param.sfo (#3405)

* Update app_content.cpp

* Use hyphenation to determine entitlement id from folder name

The original approach I took had two limitations: it relied on entitlement ids being 16 characters long, and it relied on the end of the folder name containing the entitlement. If the former wasn't the case, my code would throw an exception, while the latter would cause the DLC to not detect.
To resolve both issues, I've created a more robust algorithm based on observations from the most commonly used PS4 dumpers for modern firmware.

* Use DLC param.sfo to determine entitlement id

While the logic ends up slightly more complex, this makes the code more robust for other dumping methods/weird DLC folder names from people installing DLC manually.

* Update sceAppContentAddcontMount to properly detect additional content folders

Based on what I've done in sceAppContentInitialize, I've added code for detecting the correct folder to mount.
I've also removed the redundant check for addcont_info status, since we're marking all additional content as installed during sceAppContentInitialize

DanielSvoboda committed:
Fix sceVoicePortInfo (#3408)

The VoicePortInfo 'state' was changed from 3-RUNNING to 0-IDLE. This prevents it from getting into a loop.

kalaposfos13 committed:
Fix handling of RFC 3339 formatted dates (#3410)

StevenMiller123 committed:
Stub NOT_IN_PARTY behavior (#3409)

From what I can tell, this is valid behavior since a user could choose not to enter a party.
This fixes a potential crash on boot in Grand Theft Auto V.

kalaposfos13 committed:
Add stubbed libSceNpProfileDialog library (#3411)

* Stubbed library

* Silence sceNpProfileDialogUpdateStatus

* the loathsome clang-formatter

StevenMiller123 committed:
Add entries for . and .. in MntPoints::IterateDirectory (#3414)

Grand Theft Auto V uses sceKernelGetdents in a loop to search through the contents of /download0, but will always check the first returned directory entry regardless of what value the function returns.
As it turns out, this will never fail on real hardware because all directories have entries for . and .., while the game code throws an exception on shadPS4 because we don't emulate these entries.

rainmakerv3 committed:
Log user CPU, total RAM, and OS (#3402)

* cpu info

Update cpu_info.h

Revert "Update cpu_info.h"

This reverts commit 6db3814e7f8162ca546c33479942d7204524b281.

Revert "cpu info"

This reverts commit cfecdf694fd7bee0fdf025c3933ff7b9498a1c3f.

hardware info

* log OS

* Round RAM, add space

* switch temporarily to local fork

* Update CMakeLists.txt

* set shallow = true for submodule

* Log logical cores as well

* point submodule to shadps4 fork

StevenMiller123 committed:
Filesystem: Directory-related fixes (#3416)

* Various GetDents fixes

Fixed return and parameter types, and made the function return all the entries that will fit in nbytes during one call.

* Fstat dir stub changes

Changes the returned statistics to match what my PS4 tests generally show.

* Stat dir stub changes

To match my fstat changes

abouvier committed:
hwinfo: track the correct commit (#3419)

StevenMiller123 committed:
Memory: Align size and address in posix_munmap (#3418)

* Properly align address and size in munmap

Based on observations of FreeBSD source code, fixes a Windows-related memory issue in War Thunder (CUSA00224)

* Format len and phys_addr in mmap

This should make logs slightly easier to understand, since we format these parameters in other memory calls.

* Update memory.cpp

Missake212 committed:
Update building-windows.md (#3422)

georgemoralis committed:
New Crowdin updates (#3412)

* New translations en_us.ts (Urdu (Pakistan))

* New translations en_us.ts (Arabic)

* New translations en_us.ts (Spanish)

* New translations en_us.ts (French)

rainmakerv3 committed:
Config: toggle background controller inputs (#3424)

georgemoralis committed:
[ci skip] Qt GUI: Update Translation. (#3425)

Co-authored-by: georgemoralis <4313123+georgemoralis@users.noreply.github.com>

baggins183 committed:
Improve heuristic for attributes passed via ring buffers. (#3426)

Previously a buffer load in a vertex shader could be treated like a ring access, dropping offen vgpr and possibly asserting during resource tracking because of mismatch between types (u32x2 vs U32), caused by inconsistencies in flags (index_enable and offset_enable)

Missake212 committed:
remove wolf (#3432)

raphaelthegreat committed:
ir: Perform degamma in shader when sampler sets force_degamma (#3420)

* ir: Perform degamma in shader when sampler sets force_degamma

* specialization: Add srgb if image is sampled

Might fix cases where sampler force_degamma is used with srgb image

StevenMiller123 committed:
Libraries: Better libSceMove stubs (#3433)

* Improved libSceMove stubs

These should more accurately represent how an actual PS4 behaves when a game calls libSceMove functions while no move controllers are connected.

* Clang

* Change sceMoveGetExtensionPortInfo stub

Not entirely sure the ExtensionPortData struct is for this one, but the struct itself isn't exactly important for now anyway.

* Fix sceMoveTerm

* Update move.cpp

StevenMiller123 committed:
Libraries: Better libSceCamera stubs (#3434)

* Initial work on improved stubs

Simulates library behaviors for when a camera is not connected.

* Get* functions complete

I ended up leaving some stubs behind, but most of what I didn't do seem to be internal functions called by other libraries we'd be running HLE anyway.

* Finished stubs

Everything I can reasonably confirm the behavior of should behave as expected, emulating the behavior of a PS4 with no camera attached.

* sceCameraStart firmware check

This check only applies to higher eboot firmwares, since my previous test eboots used firmware 1.00 I completely missed this.

* sceCameraGetAutoExposureGain fix

When option is provided and valid, size is also set to 0.

* Track opened handles

Extremely basic for now, if the library was ever properly implemented this could be swapped for a map of some kind instead.

* Fix errors for sceCameraStart

The firmware-related parameter checks come after the library opened check.

* Promote sceCameraIsAttached log to info

Since you don't need to initialize the library to call this function, some games will call sceCameraIsAttached before anything else.

squidbus committed:
externals: Update MoltenVK (#3435)

Missake212 committed:
Update settings_dialog.cpp (#3436)

kalaposfos13 committed:
Implement ORBIS_NET_SO_ERROR_EX in GetSocketOptions (#3365)

rainmakerv3 committed:
Fix stop hotkey (#3438)

squidbus commented on src/common/config.cpp:
```diff
@@ -844,6 +872,8 @@ void save(const std::filesystem::path& path) {
     data["Input"]["useUnifiedInputConfig"] = useUnifiedInputConfig;
     data["Input"]["micDevice"] = micDevice;
+    data["Output"]["mainOutputDevice"] = mainOutputDevice;
+    data["Output"]["padSpkOutputDevice"] = padSpkOutputDevice;
```
Would prefer `Audio` as the category here (and above).

squidbus commented on src/core/libraries/audio/sdl_audio.cpp:
```diff
@@ -21,8 +21,29 @@ class SDLPortBackend : public PortBackend {
+            devId = SDL_AUDIO_DEVICE_DEFAULT_PLAYBACK;
+        } else {
+            try {
+                devId = static_cast<u32>(std::stoul(port_name));
```
Would make more sense to be able to specify your audio device by name rather than the device number, which may not be stable.

nickci2002 replied:
There's no function that immediately gets the device id by name in SDL. I added a check to assert the devid has the name specified in the config file. This should make it stable.

squidbus reviewed (commented):
Sorry for the delay, two comments (also needs a rebase).

kalaposfos13 committed:
Implement V_CMP_GT_U64 (#3352)

* Implement V_CMP_GT_U64

* Add GroupAny

* Use GroupAny

* Add assert

* clang

Missake212 committed:
Make shadps4 default repo for patches (#3431)

* make shadps4 default repo for patches

* if present, put shadPS4 patches on top of listview

* Update cheats_patches.cpp

---------

Co-authored-by: rainmakerv2 <30595646+rainmakerv3@users.noreply.github.com>

Missake212 committed:
Dropping "SDL audio queue backed up..." to INFO (#3437)

* Update sdl_audio.cpp

* clang (I hope)

* clang

nickci2002 committed:
Git reset to audioDev PR merge

Fixing branch graph

nickci2002 committed:
Merge pull request #3 from rainmakerv3/audioDev

Audio dev

nickci2002 committed:
Updated rainmaker's conversations

nickci2002 committed:
Updated rainmaker's conversations

nickci2002 committed:
Merge branch 'main' into DS4-Speaker-Audio-Base

nickci2002 committed:
Removed duplicate socket case in kernel/file_system.cpp

nickci2002 reviewed (commented):
See updates in sdl_audio.cpp

nickci2002 committed:
Remove duplicate Socket declaration in fs.h

nickci2002 committed:
Merge branch 'main' into DS4-Speaker-Audio-Base

nickci2002 committed:
Update settings_dialog.ui

Fixed to match original commit on head branch

nickci2002 committed:
Merge branch 'main' into DS4-Speaker-Audio-Base

nickci2002 committed:
Fixed clang-format and micDevice errors

nickci2002 committed:
Fixed clang-format and micDevice errors

nickci2002 committed:
Fixed whitespace error for clang-format sorry

nickci2002 committed:
Fixed whitespace error for clang-format sorry

nickci2002 committed:
Merge branch 'main' into DS4-Speaker-Audio-Base

nickci2002 committed:
Merge remote-tracking branch 'origin/DS4-Speaker-Audio-Base' into DS4-Speaker-Audio-Base
